### PR TITLE
feat(migrate): add migration script for Matters coupon

### PR DESCRIPTION
**Pull request by devformatters**

### CODE REVIEW

File: db/migrations/20210427123109_migrate_subscription.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  const update = async ({ table, where, data }) => {
+    const [record] = await knex(table).where(where).update(data).returning('*')
+    return record
+  }
+
+  // fetch accepted invitations
+  const invts = await knex
+    .select()
+    .from(t_invitation)
+    .where({ accepted: true })
```
Comment (robertu7) [inline db/migrations/20210427123109_migrate_subscription.js:45]: skip migrated invitations (`subscription_item_id` ) if we retry?

File: db/migrations/20210427123109_migrate_subscription.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+    const user = await findFirst({
+      table: t_user,
+      where: { ...(userId ? { id: userId } : { email }) },
+    })
+
+    if (!user) {
+      continue
+    }
+
```
Comment (robertu7) [inline db/migrations/20210427123109_migrate_subscription.js:63]: Is knex migration transactional (one failure of invitation migration will fail them all)? Shall we print debug info with masked user data (e.g.user id) on every step?

File: db/migrations/20210427123109_migrate_subscription.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+          provider: 'matters',
+          provider_subscription_id: v4(),
+          state: 'trialing',
+          user_id: user.id,
+        },
+      })
+    }
+
+    // Step 3: create Matters subscription item
+    const { id, circle_id, ...rest } = item
```
Comment (robertu7) [inline db/migrations/20210427123109_migrate_subscription.js:112]: explicit better than implicit (`...rest`)?

**Review by robertu7 — CHANGES_REQUESTED** [review]

**Review by guoliu — APPROVED** [review]
Let's try it 👍

File: db/migrations/20210427123109_migrate_subscription.js (hunk 1 above)
Comment (devformatters) [reply db/migrations/20210427123109_migrate_subscription.js:32]: Added.

File: db/migrations/20210427123109_migrate_subscription.js (hunk 2 above)
Comment (devformatters) [reply db/migrations/20210427123109_migrate_subscription.js:60]: Added logger and break original script into two migration.

File: db/migrations/20210427123109_migrate_subscription.js (hunk 3 above)
Comment (devformatters) [reply db/migrations/20210427123109_migrate_subscription.js]: Changed.

File: db/migrations/20210427142108_migrate_stripe_subscription.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+      if (!stripeSubscription) {
+        console.log('Stripe subscription not found')
+        continue
+      }
+
+      if (stripeSubscription.canceled_at > 0) {
+        console.log('Stripe subscription arleady canceled')
+        continue
+      }
```
Comment (robertu7) [inline db/migrations/20210427142108_migrate_stripe_subscription.js:68]: also need to check if subscription items already been deleted?

**Review by robertu7 — APPROVED** [review]

File: db/migrations/20210427142108_migrate_stripe_subscription.js (hunk 4 above)
Comment (devformatters) [reply db/migrations/20210427142108_migrate_stripe_subscription.js:68]: Nice point. Added.

**Review by robertu7 — APPROVED** [review]

**Pull request merged**